frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -137,6 +137,10 @@ def load_order
       redirect_to(spree.cart_path) && return unless @order
     end
 
+    # Allow the customer to only go back or stay on the current state
+    # when trying to change it via params[:state]. It's not allowed to
+    # jump forward and skip states (unless #skip_state_validation? is
+    # truthy).
     def set_state_if_present
```

Comment:
What about also changing this method name? We could keep this one so that it only prints the deprecation message and call the new one, WDYT?

Reply:
I'd be very much in favor of doing so! I'll make a note to do another PR with the full deprecation work.

Reply:
https://github.com/solidusio/solidus/pull/3496 ✨